openedx/features/badging/helpers.py

```diff
@@ -65,3 +103,28 @@ def add_badge_earned_date(course_id, course_badges, earned_badges):
             if badge['id'] == earned_badge.badge_id and course_id == earned_badge.course_id:
                 # earned date indicate badge is earned
                 badge['date_earned'] = earned_badge.date_earned
+
+
+def filter_earned_badge_by_joined_team(user, course, earned_badges):
+    """
+    This method filter badges earned, in a course, by team joined by user
+    :param user: Current logged-in user
+    :param course: Course, user enrolled in
+    :param earned_badges: All badges earned in a course
+    :return: All badges earned in a course, specific to joined team
+    """
+    course_team = CourseTeam.objects.filter(course_id=course.id, users=user).values('id').first()
+    if course_team:
+        team_group_chat = TeamGroupChat.objects.filter(team_id=course_team['id']).values('room_id').first()
+
+        if not team_group_chat:
+            error = FILTER_BADGES_ERROR.format(team_id=course_team['id'])
+            log.exception(error)
+            raise Exception(error)
+
+        # filter earned badges for joined team only
+        earned_badges = [
+            earned_badge for earned_badge in earned_badges if
+            earned_badge.community_id == team_group_chat['room_id']
+        ]
+    return course_team, earned_badges
```

Comment:
If `course_team` is not found it means, user has not joined any team in particular course.

Since this function is only for filtering earned badges, 'course team not found' case is handled, from where is function is called.

Reply:
In my mind the flow indicates that if we're fetching team here, we should raise the exception from here otherwise it should be passed. Does that make sense?